src/phoenix/server/api/types/pagination.py

```diff
@@ -56,6 +60,73 @@ class Edge(Generic[GenericType]):
 CURSOR_PREFIX = "connection:"
 
 
+class SortableFieldType(Enum):
+    STRING = auto()
+    INT = auto()
+    FLOAT = auto()
+    DATETIME = auto()
+
+
+@dataclass
+class SortableField:
+    type: SortableFieldType
+    value: SortableFieldValue
+
+    def stringify_value(self) -> str:
+        if isinstance(self.value, str):
+            return self.value
+        if isinstance(self.value, (int, float)):
+            return str(self.value)
+        if isinstance(self.value, datetime):
+            return self.value.isoformat()
+        assert_never(self.type)
+
+    @classmethod
+    def from_stringified_value(
+        cls, type: SortableFieldType, stringified_value: str
+    ) -> "SortableField":
+        value: SortableFieldValue
+        if type is SortableFieldType.STRING:
+            value = stringified_value
+        elif type is SortableFieldType.INT:
+            value = int(stringified_value)
+        elif type is SortableFieldType.FLOAT:
+            value = float(stringified_value)
+        elif type is SortableFieldType.DATETIME:
+            value = datetime.fromisoformat(stringified_value)
+        else:
+            assert_never(type)
+        return cls(type=type, value=value)
+
+
+@dataclass
+class NodeIdentifier:
+    rowid: int
+    sortable_field: Optional[SortableField] = None
+
+    _DELIMITER: ClassVar[str] = ":"
```

Comment:
can be a constant outside of the class

Reply:
Think I'd like to keep it encapsulated within the class since this is the only place is it referenced. Mentally, I am thinking of this class as containing all the internal details of serializing and deserializing cursors, so having this constant encapsulated makes it harder for those details to leak outside the class.